webapi/Controllers/DocumentController.cs

```diff
@@ -125,12 +86,82 @@ public bool ContentSafetyStatus()
+    [Route("document/importstatus")]
     [HttpPost]
     [ProducesResponseType(StatusCodes.Status200OK)]
     [ProducesResponseType(StatusCodes.Status400BadRequest)]
-    public async Task<IActionResult> ImportDocumentsAsync(
-        [FromServices] IKernel kernel,
+    public async Task<IActionResult> DocumentStatusAsync(
+        [FromServices] ISemanticMemoryClient memoryClient,
+        [FromServices] IHubContext<MessageRelayHub> messageRelayHubContext,
+        [FromBody] DocumentStatusForm documentStatusForm)
+    {
+        try
+        {
+            await this.ValidateDocumentStatusFormAsync(documentStatusForm);
+        }
+        catch (ArgumentException ex)
+        {
+            return this.BadRequest(ex.Message);
+        }
+
+        var userId = this._authInfo.UserId;
+        var chatId = documentStatusForm.ChatId.ToString();
+        var targetCollectionName = documentStatusForm.DocumentScope == DocumentScopes.Global
+            ? this._options.GlobalDocumentCollectionName
+            : this._options.ChatDocumentCollectionNamePrefix + chatId;
+
+        var statusResults = await QueryAsync().ToArrayAsync();
+
+        //// Broadcast the document status event to other users.
+        //if (documentStatusForm.DocumentScope == DocumentScopes.Chat) $$$ TODO - OPEN DESIGN POINT
+        //{
+        //    await messageRelayHubContext.Clients.Group(chatId)
+        //        .SendAsync(ReceiveMessageClientCall, chatId, userId, chatMessage); // $$$ STATUS
+
+        //    return this.Ok(chatMessage);
+        //}
+
+        //await messageRelayHubContext.Clients.All.SendAsync(
+        //    GlobalDocumentUploadedClientCall, // $$$ STATUS
+        //    documentMessageContent.ToFormattedStringNamesOnly(),
+        //    documentStatusForm.UserName
+        //);
+
+        return this.Ok("Documents status reported.");
+
+        async IAsyncEnumerable<StatusResult> QueryAsync()
+        {
+            foreach (var documentReference in documentStatusForm.FileReferences)
+            {
+                var status = await memoryClient.GetDocumentStatusAsync(targetCollectionName!, documentReference!);
+                if (status == null)
+                {
+                    yield return new StatusResult(targetCollectionName, documentReference);
+                }
+                else
+                {
+                    yield return new StatusResult(targetCollectionName, documentReference)
+                    {
+                        IsStarted = true,
+                        IsCompleted = status.Completed,
+                        LastUpdate = status.LastUpdate,
+                        Keys = Array.Empty<string>(), // $$$ DCR MEMORY | DCR CHAT
+                        Tokens = 0, // $$$ DCR MEMORY | DCR CHAT
+                    };
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// Service API for importing a document.
+    /// </summary>
+    [Route("document/import")]
```

Comment:
We should really think through these route changes to make sure they align with Microsoft's REST API standards. Route changes are breaking changes, so we need to make sure we do them right and do them once. We also need to provide mitigation steps for our customers to minimize downtime.

There are additional route changes we are planning to make (see issue #141) so it might be good to do them all together, so we don't break customers multiple times. Should discuss with @alliscode

Reply:
Thanks for the call-out on #141.  In terms of customer impact, I'd expect that app and api would be updated together.  (I mean we regularly make changes where an app deployment alone would break w/o updating API.)

We certainly need to confirm our thoughts around this...totally agree.